install.go

```diff
@@ -542,3 +547,65 @@ func completeFileName(dir, name string) (string, error) {
 
 	return "", nil
 }
+
+// checkGpgKeys iterates through the keys listed in the PKGBUILDs and if needed,
+// asks the user whether yay should try to import them.
+func checkGpgKeys(srcinfos map[string]*gopkg.PKGBUILD) error {
+	// keys maps a given PGP with a boolean to indicate whether we would need to
+	// import this particular key.
+	keys := map[string]bool{}
+	// args will be usde to check all the keys at once.
+	args := []string{"--list-keys", "--with-sig-check"}
+
+	for _, pkgbuild := range srcinfos {
+		for _, key := range pkgbuild.Validpgpkeys {
+			// Using ToUpper to make sure case won't be an issue.
+			keys[strings.ToUpper(key)] = false
+			// For this check, it is OK if there are repeated keys.
+			args = append(args, key)
+		}
+	}
+
+	cmd := exec.Command(config.GpgBin, args...)
+	err := cmd.Run()
+
+	if err != nil {
+		// There was an issue with at least one key, so let's find out which
+		// ones gave us problems and we can offer to import them.
+		problematic := []string{}
+		for key := range keys {
+			args = []string{"--list-key", key}
+			cmd = exec.Command(config.GpgBin, args...)
+			err = cmd.Run()
+			if err != nil {
+				problematic = append(problematic, key)
+				keys[key] = true
+			}
+		}
+		keyStr := "key"
+		if len(problematic) > 1 {
+			keyStr = "keys"
+		}
+		question := fmt.Sprintf("Problem detected with the following %s: %q. Import?", keyStr, strings.Join(problematic[:], ","))
```

Comment:
This is proving to be way too long of a line this should probably be done like:

```
GPG keys need importing:
    key1
    key2
    key3
Import? [Y/n]
```
Additionally it might be nice to say what pkgbuild wants to import what key although I am unsure how to format that currently.

This is a formatting issue though, you shouldn't need to worry about it until implementation is finished.